### PR TITLE
ddl: index name should not be `primary` (#3834)

### PR DESCRIPTION
**Pull request by jackysp**
Fix issue #3834 
PTAL @zimulala @XuHuaiyu

### CODE REVIEW

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -498,6 +498,10 @@ func setEmptyConstraintName(namesMap map[string]bool, constr *ast.Constraint, fo
 		colName := constr.Keys[0].Column.Name.L
 		constrName := colName
 		i := 2
+		if strings.ToUpper(constrName) == mysql.PrimaryKeyName {
```
Comment (breeswish) [inline ddl/ddl_api.go]: Use `strings.EqualFold`

Comment (breeswish) [inline ddl/ddl_api.go]: Why do we need to set extra rule when its name is `PRIMARY` here?

File: plan/validator.go [hunk 2]
```diff
@@ -315,9 +315,7 @@ func (v *validator) checkCreateIndexGrammar(stmt *ast.CreateIndexStmt) {
 		v.err = ddl.ErrWrongTableName.GenByArgs(tName)
 		return
 	}
-	// We do not check column name here, due to MySQL returns "ERROR 1072 (42000): Key column 'a ' doesn't exist in table",
```
Comment (XuHuaiyu) [inline plan/validator.go]: why remove this comment?

File: ddl/ddl_api.go (hunk 1 above)
Comment (jackysp) [inline ddl/ddl_api.go]: Because MySQL will append `_{num}` to anonymous index name if index column name is `primary`.

File: ddl/ddl_db_test.go [hunk 3]
```diff
@@ -323,6 +323,14 @@ func (s *testDBSuite) testAddAnonymousIndex(c *C) {
...
 	s.mustExec(c, "alter table t_anonymous_index add index c3 (C3)")
 	s.mustExec(c, "alter table t_anonymous_index drop index C3")
+	// for anonymous index with column name `primary`
+	s.mustExec(c, "create table t_primary (`primary` int, key (`primary`))")
+	t = s.testGetTable(c, "t_primary")
+	c.Assert(t.Indices()[0].Meta().Name.String(), Equals, "primary_2")
+	s.mustExec(c, "create table t_primary_2 (`primary` int, key primary_2 (`primary`), key (`primary`))")
+	t = s.testGetTable(c, "t_primary_2")
+	c.Assert(t.Indices()[0].Meta().Name.String(), Equals, "primary_2")
+	c.Assert(t.Indices()[1].Meta().Name.String(), Equals, "primary_3")
```
Comment (XuHuaiyu) [inline ddl/ddl_db_test.go]: we can add a test case 
```sql
create table t (`primary_2` int, key(`primary_2`), `primary` int, key(`primary`));
```

File: plan/validator.go (hunk 2 above)
Comment (jackysp) [inline plan/validator.go]: It is a little weird to keep this single comment here.

**Comment by XuHuaiyu** [comment]
LGTM 
please resolve the conflicts.

File: ddl/ddl_api.go (hunk 1 above)
Comment (jackysp) [inline ddl/ddl_api.go]: Done!

File: ddl/ddl_db_test.go (hunk 3 above)
Comment (jackysp) [inline ddl/ddl_db_test.go]: Done!

**Comment by shenli** [comment]
It seems that this PR is not related with issue  #3834.

**Comment by shenli** [comment]
LGTM

**Comment by jackysp** [comment]
It is actually #3843 . Sorry for my mistake.

**Pull request merged**